src/Raven.Server/Documents/PeriodicBackup/PeriodicBackupRunner.cs

```diff
@@ -1012,7 +1057,31 @@ public RunningBackup OnGoingBackup(long taskId)
 
         public Dictionary<string, long> GetLastProcessedTombstonesPerCollection(ITombstoneAware.TombstoneType tombstoneType)
         {
-            var minLastEtag = GetMinLastEtag();
+            var minLastEtag = GetMinLastEtag(out var taskIdsStatusesToDelete);
+
+            if (taskIdsStatusesToDelete != null && taskIdsStatusesToDelete.Count > 0)
+            {
+                try
+                {
+                    if (_serverStore.DatabaseInfoCache.BackupStatusStorage.DeleteBackupStatusesByTaskIds(_database.Name, _serverStore._env.Base64Id, taskIdsStatusesToDelete) == false)
```

Comment:
You are in a `Task` here (the tombstone cleaner runs as task) and `DeleteBackupStatusesByTaskIds` opens a write tx, so it might be blocking.

Can we move this to be part of the raft command for the updating the backup status.

Reply:
I also do not like a side effect of this method, get method is opening a write tx

Reply:
moved to UpdatePeriodicBackupStatusCommand